File: client/src/types/order.ts
export interface OrderCategory {
  categoryId: number;
  categoryName: string;
}
export interface SkinType {
  skinTypeId: number;
  skinTypeName: string;
}

export interface SkinTypeWrapper {
  $id: string;
  $values: SkinType[];
}
export interface OrderDetail {
  orderDetailId: number;
  productId: number;
  productName: string;
  size: string;
  quantity: number;
  price: number;
  discount: number;
  productImage?: string;
  category: OrderCategory;
  skinTypes: SkinTypeWrapper;
}
export interface OrderDetailWrapper {
  $id: string;
  $values: OrderDetail[];
}
export interface Order {
  orderId: number;
  orderCode: string;
  fullName: string;
  address: string;
  phoneNumber: string;
  voucher: Voucher | null;
  shippingPrice: number;
  totalAmount: number;
  paymentMethodName: string;
  status: string;
  createdDate: string;
  details: OrderDetail[];
}
export interface PurchaseDetail {
  $id: string;
  orderId: number;
  orderCode: string;
  fullName: string;
  address: string;
  shippingPrice: number;
  paymentMethodName: string;
  voucher: Voucher | null;
  phoneNumber: string;
  totalAmount: number;
  status: string;
  createdDate: string;
  details: OrderDetailWrapper;
}
export interface Voucher {
  $id: string;
  voucherId: number;
  voucherName: string;
  voucherCode: string;
  description: string;
  discountAmount: number | 0;
  startDate: string;
  endDate: string;
  status: boolean;
  minimumPurchase: number;
}

File: server/dotnet/ThanhCometics.Data/Entities/Order.cs
using System;
using System.Collections.Generic;

namespace ThanhCometics.Repository.Entities;

public partial class Order
{
    public int OrderId { get; set; }

    public string? OrderCode { get; set; }

    public decimal? TotalAmount { get; set; }

    public string FullName { get; set; } = null!;

    public string Address { get; set; } = null!;

    public string PhoneNumber { get; set; } = null!;

    public string Status { get; set; } = null!;

    public DateTime? CreatedDate { get; set; }

    public decimal ShippingPrice { get; set; }

    public int CustomerId { get; set; }

    public int? VoucherId { get; set; }

    public int PaymentMethodId { get; set; }

    public virtual Customer Customer { get; set; } = null!;

    public virtual ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();

    public virtual PaymentMethod PaymentMethod { get; set; } = null!;

    public virtual ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();

    public virtual Voucher? Voucher { get; set; }
}
